### PR TITLE
add core.request.get_scheme/host/port/http_version 

### PR DESCRIPTION
**Pull request by taotao2014**
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

**Comment by membphis** [comment]
we should keep those codes.

image

**Comment by taotao2014** [comment]
What can i do ? Adding it has conflicts

**Comment by membphis** [comment]
> What can i do ? Adding it has conflicts

if you do not know how to fix it, you can create a new PR with your new code. then we can close this stale PR.

**Comment by membphis** [comment]
> What can i do ? Adding it has conflicts

I helped you fix this case. you can continue your job.

File: t/core/request.t [hunk 1]
```diff
@@ -206,3 +206,151 @@ X-Forwarded-For: 10.0.0.1
...
+        }
+    }
+--- request
+GET /hello
+--- more_headers
+X-Real-IP: 10.0.0.1
+--- response_body
+1.1
+--- no_error_log
+[error]
```
Comment (membphis) [inline t/core/request.t:356]: please add a blank line at the end of file

Comment (taotao2014) [reply t/core/request.t]: completed

**Pull request merged**